<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/ycn/damiao_ws/src/damiao_ros2_control/urdf/damiao_motor.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="damiao_motor">
  <!-- ros2_control 硬件接口配置 -->
  <ros2_control name="DamiaoSystem" type="system">
    <hardware>
      <plugin>damiao_ros2_control/DamiaoSystem</plugin>
  
      <param name="serial_port">/dev/ttyACM0</param>
      <param name="baud_rate">921600</param>

    </hardware>

    <joint name="motor1_joint">

      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
  
      <command_interface name="position"/>
      <command_interface name="velocity"/>
    </joint>
  </ros2_control>

  <!-- ====================== 底座：竖直杆（base_link） ====================== -->
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.000253477290225107 -1.40368868973369E-05 0.0241187399448777"
        rpy="0 0 0" />
      <mass
        value="0.308025461282533" />
      <inertia
        ixx="9.03921486870941E-05"
        ixy="8.15902115481825E-08"
        ixz="1.25501300090895E-07"
        iyy="9.03283141997523E-05"
        iyz="1.59574505477921E-09"
        izz="0.000126078332148332" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://damiao_ros2_control/urdf/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.2 0.2 0.2 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://damiao_ros2_control/urdf/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="motor1_link">
    <inertial>
      <origin
        xyz="-0.0470262679774121 0.0169861743814382 -0.005"
        rpy="0 0 0" />
      <mass
        value="0.0446211275016188" />
      <inertia
        ixx="1.1235904408583E-05"
        ixy="1.81462262697607E-05"
        ixz="3.16942446103158E-22"
        iyy="5.49192516191763E-05"
        iyz="-1.49389941997121E-22"
        izz="6.54114705693989E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://damiao_ros2_control/urdf/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://damiao_ros2_control/urdf/link_1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="motor1_joint"
    type="revolute">
    <origin
      xyz="0.00017354 -1.2493E-05 0.056"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="motor1_link" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="10"
      velocity="10" />
  </joint>
</robot>
